components/rangeslider/script.js

```diff
@@ -0,0 +1,475 @@
+        return polContainer;
+    }
+
+    /**
+     * Creates the thumb element
+     * @param {number || string} value - the initial value of the thumb
+     */
+    buildThumb(value) {
+        const thumb = document.createElement('div');
+        thumb.classList.add(`${this.orientation}-rangeslider-thumb`);
+        thumb.textContent = value;
+        this.rangeslider.appendChild(thumb);
+    }
+
+    /**
+     * Gets the rangeslider dimensions
+     * @returns {DOMRect}
+     */
+    getRangeSliderSize() {
+        return this.querySelector(`.${this.orientation}-rangeslider-wrapper`).getBoundingClientRect();
+    }
+
+    /**
+     * Sets the min and max value of the rangeslider
+     */
+    setMinAndMax() {
+        //if we have an array we set the min and max values to the first and last entry of the array
+        if (this.isArray) {
+            this.min = this._values[0];
+            this.max = this._values[this._values.length - 1];
+            return;
+        }
+
+        this.min = this.getAttribute('min') || 0;
+        this.max = this.getAttribute('max') || 100;
+    }
+
+    /**
+     * Sets the value of the handles
+     */
+    setHandleValues() {
+        //if there are two handles we set their values to the min and max
+        if (this.twoHandles) {
+            this.value = [this.min, this.max];
+            return;
+        }
+
+        this.value = [this.getAttribute('value')] || [this.min];
+        //checks if the value provided is less than the min or more than the max and sets it to the minimum value
+        this.value = this.min <= this.value && this.max >= this.value ? [this.value] : [this.min];
+    }
+
+    /**
+     * Calculates the value of the handle based on the position of the handle
+     * @param {number} percent - the percent position
+     * @returns {number} - the value of the handle
+     */
+    calculateHandleValue(percent) {
+        return parseFloat(parseInt(this.min) + (this.max - this.min) * (percent / 100));
+    }
+
+    /**
+     * Attaches the event listener
+     */
+    attachEventListeners() {
+        this.querySelector(`.${this.orientation}-rangeslider-wrapper`).addEventListener('mousedown', this.onMouseDown);
+    }
+
+    /**
+     * Updates the positions of the handles and the width of the bar
+     * @param {number} percent
+     * @param {number} index - the index of the handle we want to update
+     */
+
+    updateSliderPosition(percent, index) {
+        //The percent of the step that is set, if the values are an array, the step is between each array value
+        const percentStep = this.isArray
+            ? 100 / (this._values.length - 1)
+            : Rangeslider.valueToPercent(this.step, this.min, this.max);
+
+        //the range which should be clamped
+        let clampRange = [0, 100];
+        //if we have two handles we need to clamp each so that it doesn't pass beyond the other handle
+        if (this.twoHandles && this.handle[1].style[this.units.position]) {
```

Comment:
Because when we set the positions of the handles on setup, there isn't a position for handle[1] or handle[0] yet. So we need to clamp them between 0 and 100 since they are on both ends. After that they will have set positions so we can clamp them each handle to be between the end and the other handle.

This is why I have chosen to handle[1] because if it was handle[0] then in the setup on handle[1] it will still enter the if statement as the other handle already has it's position set. An alternative would be to use this.handle[activeIndex]. If it makes more sense to you, I'll change it.